client/src/pages/Interview.js

```diff
@@ -101,27 +100,44 @@ const Interview = (props) => {
   const interviewId = props.match.params.id;
 
   const { socket } = useContext(SocketContext);
-  const { difficulty } = useContext(UserContext);
 
   useEffect(() => {
     barRef.current && setBarHeight(barRef.current.clientHeight);
   }, [barRef]);
 
   useEffect(() => {
-    const getQuestion = async () => {
-      const res = await fetch("/api/question", {
-        method: "post",
-        body: JSON.stringify({ difficulty }),
-        headers: {
-          "Content-Type": "application/json",
-        },
-      });
-      const question = await res.json();
-      setQuestion(question);
+    let interviewObjFromDB;
+    const getInterviewAndQuestion = async () => {
+      const getInterview = async () => {
+        const res = await fetch(`/api/interview/${interviewId}`, {
+          method: "get",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        });
+
+        interviewObjFromDB = await res.json();
+      };
+
+      const getQuestion = async () => {
+        const res = await fetch(
+          `/api/question/${interviewObjFromDB.questions[0]}`,
+          {
+            method: "get",
+            headers: {
+              "Content-Type": "application/json",
+            },
+          }
+        );
```

Comment:
would it make sense to combine these on the backend? So that instead of making two calls, the backend just returns the interview as well as the question details. This can be done with a separate route or by modifying this route depending on how it's used elsewhere or by adding a parameter like `includeQuestion` or something

Reply:
Yes, That is a good idea. I will define a new route on backend that get the interviewId & returns the question of that interviewID.